application/server/configuration.go

```diff
@@ -192,81 +262,73 @@ func updateFolderConfig(c *config.Config, settings types.Settings, logger *zerol
-	} else {
-		// Folder org matches global org after LDX-Sync - clear it and set user flag
-		folderConfig.PreferredOrg = ""
-		folderConfig.OrgSetByUser = userExplicitlySetOrg || c.Organization() != ""
+	// ReferenceFolderPath change
+	if oldReferenceFolderPath != storedConfig.ReferenceFolderPath {
+		go sendConfigChangedAnalyticsEvent(c, configReferenceFolderPath, oldReferenceFolderPath, storedConfig.ReferenceFolderPath, path, triggerSource)
 	}
 
-	folderConfig.OrgMigratedFromGlobalConfig = true
-}
+	// ScanCommandConfig change
+	if !reflect.DeepEqual(oldScanCommandConfig, storedConfig.ScanCommandConfig) {
+		go sendMapConfigChangedAnalytics(c, "scanCommandConfig", oldScanCommandConfig, storedConfig.ScanCommandConfig, path, triggerSource)
+	}
 
-func setOrgFromLdxSync(c *config.Config, storedConfig *types.FolderConfig, notifier noti.Notifier) (newOrgIsDefault bool) {
-	logger := c.Logger().With().Str("method", "updateAndSendFolderConfigs").Logger()
+	// PreferredOrg change
+	if oldOrg != storedConfig.PreferredOrg && storedConfig.PreferredOrg != "" {
+		go sendConfigChangedAnalyticsEvent(c, configPreferredOrg, oldOrg, storedConfig.PreferredOrg, path, triggerSource)
+	}
 
-	path := storedConfig.FolderPath
+	// OrgMigratedFromGlobalConfig change
+	if oldOrgMigratedFromGlobalConfig != storedConfig.OrgMigratedFromGlobalConfig {
+		go sendConfigChangedAnalyticsEvent(c, configOrgMigratedFromGlobalConfig, oldOrgMigratedFromGlobalConfig, storedConfig.OrgMigratedFromGlobalConfig, path, triggerSource)
+	}
 
-	newOrg, err := ldx_sync_config.ResolveOrganization(c.Engine().GetConfiguration(), c.Engine(), &logger, string(path), storedConfig.PreferredOrg)
-	if err != nil {
-		logger.Err(err).Msg("unable to resolve organization")
-		notifier.SendShowMessage(sglsp.MTError, err.Error())
-	} else {
-		storedConfig.PreferredOrg = newOrg.Id
+	// OrgSetByUser change
+	if oldOrgSetByUser != storedConfig.OrgSetByUser {
+		go sendConfigChangedAnalyticsEvent(c, configOrgSetByUser, oldOrgSetByUser, storedConfig.OrgSetByUser, path, triggerSource)
 	}
-	newOrgIsDefaultPtr := newOrg.IsDefault
-	return newOrgIsDefaultPtr != nil && *newOrgIsDefaultPtr
 }
 
-func updateAuthenticationMethod(c *config.Config, settings types.Settings) {
+func folderConfigsOrgSettingsEqual(folderConfig types.FolderConfig, storedConfig *types.FolderConfig) bool {
+	return folderConfig.PreferredOrg == storedConfig.PreferredOrg &&
+		folderConfig.OrgSetByUser == storedConfig.OrgSetByUser &&
+		folderConfig.OrgMigratedFromGlobalConfig == storedConfig.OrgMigratedFromGlobalConfig
+}
+
+func updateAuthenticationMethod(c *config.Config, settings types.Settings, triggerSource string) {
 	if types.EmptyAuthenticationMethod == settings.AuthenticationMethod {
 		return
 	}
 
+	oldValue := c.AuthenticationMethod()
 	c.SetAuthenticationMethod(settings.AuthenticationMethod)
 	di.AuthenticationService().ConfigureProviders(c)
+
+	if oldValue != settings.AuthenticationMethod {
+		if c.IsLSPInitialized() {
```

Comment:
in general, this check could always be integrated into one if (see above & below)